lib/core/src/client/preview/start.js

```diff
@@ -52,10 +52,20 @@ export default function start(render, { decorateStory } = {}) {
   const { clearDecorators } = clientApi;
   const configApi = new ConfigApi({ clearDecorators, ...context });
 
+  let prevKind;
+  let prevStory;
+
   // initialize the UI
   const renderUI = () => {
     if (isBrowser) {
       render(context);
+      const { selectedKind, selectedStory } = reduxStore.getState();
+      if (selectedKind !== prevKind || selectedStory !== prevStory) {
```

Comment:
Perhaps we shouldn't call into `render()` at all it if they are the same. I'm pretty sure all frameworks end up making this same check (although Angular ends up re-rendering, it is not clear why)

Reply:
Makes sense
@igor-dv @alterx WDYT?

Reply:
There can be also `state.error`

Reply:
Yes, good point. This is something I've wanted to refactor out anyway, I will get it in a separate PR.